### ISSUE TITLE
lerna version --force modifier (feature request)

### ISSUE THREAD

**Issue by nmccready** [lerna/lerna#2069]
**TLDR**: A flag or modifier to skip checks for looking for lerna packages changes is desirable. Which would allow a version bump to occur regardless of changes or not.

\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

Unlike `npm version` `lerna version` has smarts in it which prevent versioning of any bump of any of the underlying packages unless there are real changes. 

Now in the node world I agree that this is fine and dandy as most of us like to operate off of master. But by not making this optional it is making it virtually impossible to cleanly create custom `SNAPSHOT` version number for developers being ruled by a Java overlord.

Example of the problem:

Start at version `0.1.1`

```sh
# good to go we have a real patch version
yarn lerna version patch
```

now at `0.1.2`

```sh
# increment patch to become current SNAPSHOT to make java lead happy
yarn lerna version patch --no-git-tag-version # fails, no changes still on 0.1.2
yarn lerna version ${NEW_VERSION}-SNAPSHOT --no-git-tag-version # fails, no changes still on 0.1.2
```

Possible solution:
Back at 0.1.1

Example of `--force`:

```sh
# good to go we have a real patch version
yarn lerna version patch
```

now at `0.1.2`

```sh
# increment patch to become current SNAPSHOT to make java lead happy
yarn lerna version patch --force --no-git-tag-version # 0.1.2
yarn lerna version ${NEW_VERSION}-SNAPSHOT --force --no-git-tag-version # 0.1.2-SNAPSHOT
```

Apologies if there is a way around this that I missed. But this line check appears to me that there is no way around it.

https://github.com/lerna/lerna/blob/master/commands/version/index.js#L204

**Comment by evocateur** [comment]
[`--force-publish`](https://github.com/lerna/lerna/blob/master/commands/version/README.md#--force-publish)?

**Comment by travigd** [comment]
@evocateur That doesn't seem to work (maybe an issue with the combination of flags?).
```
travisd-mbp:packages travisd$ npm run lerna -- version --no-git-tag-version --no-push --exact --force-publish 0.10.0

> root@ lerna /Users/travisd/Projects/WebIO-asdf/packages
> lerna "version" "--no-git-tag-version" "--no-push" "--exact" "--force-publish" "0.10.0"

lerna notice cli v3.16.4
lerna info current version 0.8.8
lerna notice FYI git repository validation has been skipped, please ensure your version bumps are correct
lerna WARN force-publish 0.10.0
lerna info Looking for changed packages since v0.8.8
lerna success No changed packages to version 
```

For context, I'm trying to use Lerna only for the purpose of keeping versions in sync and publishing to NPM - I really don't want it doing anything with git (there's a separate process for that - the JS packages are considered secondary).

**Comment by travigd** [comment]
Ope, figured out the issue above was that `--force-publish 0.10.0` was interpreted as `--force-publish=0.10.0` which only applied the force directive to packages that matched the glob `0.10.0`.

I will say that this flag does seem a bit misnamed since it has nothing to do with actually publishing.

**Comment by nmccready** [comment]
I tested 

> [`--force-publish`](https://github.com/lerna/lerna/blob/master/commands/version/README.md#--force-publish)?

and it worked perfectly for me. Thank you for the quick fix.

**Comment by idoo** [comment]
for me it returns `lerna success No changed packages to version `

**Comment by KevinGhadyani-Okta** [comment]
I'm having the same issue now in 2023 that it says there are no packages to version.
